pkg/validation/flags.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"path/filepath"
+	"strconv"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+type WebhookOpts struct {
+	listenHost string
+	listenPort int
+	certDir    string
+	certName   string
+	keyName    string
+	// Deprecated fields to be removed
+	ListenAddress string
+	CertFile      string
+	KeyFile       string
+}
+
+func (opts *WebhookOpts) AddFlags(fs *flag.FlagSet, includeDeprecatedFlags bool) {
+	if includeDeprecatedFlags {
+		fs.StringVar(&opts.ListenAddress, "seed-admissionwebhook-listen-address", ":8100", "The listen address for the seed amission webhook (Deprecated)")
+		fs.StringVar(&opts.CertFile, "seed-admissionwebhook-cert-file", "", "The location of the certificate file (Deprecated)")
+		fs.StringVar(&opts.KeyFile, "seed-admissionwebhook-key-file", "", "The location of the certificate key file (Deprecated)")
+	}
+	fs.StringVar(&opts.listenHost, "admissionwebhook-listen-host", "", "The listen host for the seed amission webhook")
+	fs.IntVar(&opts.listenPort, "admissionwebhook-listen-port", 8100, "The listen port for the seed amission webhook")
```

Comment:
FWIW, I kinda like specifying the address to listen on in one single CLI flag, instead of having to split it into 2 fields. Did you do this to save the call the SplitHostPort() ?

Reply:
Not really, just a matter of taste I think